Allow fillAlpha to be set on a per-Series basis when fillGraph option is used.

t-lynch opened the pull request:
Changed the way that the fillAlpha option is processes in the _fillPlotter function in order to allow for per-series configuration.   Previously when fillGraph was set, only the global option was recognized.

danvk commented:
Thanks! Mind adding a quick unit test for this?

t-lynch commented:
Sure.   It may take a day or so before I can get to it.    I did do extensive testing on it in the project I'm using Dygraphs for, and once I push those into production on my end I'll revisit the unit test.

danvk commented:
Sounds good. We'll wait for a tested PR.
On Wed, May 27, 2015 at 4:28 PM Tom Lynch <notifications@github.com> wrote:

> Sure. It may take a day or so before I can get to it. I did do extensive
> testing on it in the project I'm using Dygraphs for, and once I push those
> into production on my end I'll revisit the unit test.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/danvk/dygraphs/pull/615#issuecomment-106062688>.
>

t-lynch commented:
Ok, that didn't take nearly as long as I thought it might.   Thankfully the instructions were very clear and concise.

Hopefully I took the right approach here - I added a new test to show the different ways the fillAlpha value might be populated,
* unspecified (use default value)
* globally
* per series for all series
* mix of global and per series values

I also amended the existing auto_test for the per-series options and added a new test which ensures that the per-series values are in fact used when filling the series.

Let me know if there is anything else you need to verify this PR.

danvk commented:
Merged, thanks for adding the unit test!
